test/core/end2end/tests/hpack_size.cc

```diff
@@ -386,11 +392,22 @@ void hpack_size(grpc_end2end_test_config config) {
                                           1000, 32768, 4 * 1024 * 1024};
   size_t i, j;
 
+  std::vector<std::shared_ptr<absl::Notification>> dones;
   for (i = 0; i < GPR_ARRAY_SIZE(interesting_sizes); i++) {
     for (j = 0; j < GPR_ARRAY_SIZE(interesting_sizes); j++) {
-      test_size(config, interesting_sizes[i], interesting_sizes[j]);
+      auto done = std::make_shared<absl::Notification>();
+      dones.push_back(done);
+      grpc_event_engine::experimental::GetDefaultEventEngine()->Run(
+          [done, config, i, j] {
+            test_size(config, interesting_sizes[i], interesting_sizes[j]);
+            done->Notify();
+          });
     }
   }
+  for (auto& done : dones) {
+    done->WaitForNotification();
+  }
+  grpc_event_engine::experimental::ResetDefaultEventEngine();
```

Comment:
I think we'll need to revisit where we've called `ResetDefaultEventEngine` later on. This is a necessary evil at the moment.

Reply:
ack